src/biokbase/narrative/tests/test_jobcomm.py

```diff
@@ -99,153 +104,236 @@
-        elif isinstance(req, dict):
-            request = req["content"]["data"]
-            source = req["content"]["data"]["request_type"]
-        elif isinstance(req, str):
-            request = req
-            source = req
+def check_job_id__no_job_test(job_comm: JobComm, request_type: str) -> None:
+    """Check that the correct error is raised with no job provided."""
+    for job_id in [None, ""]:
+        job_comm._comm.clear_message_cache()
+        req_dict = make_comm_msg(request_type, {JOB_ID: job_id})
+        req = make_job_request(request_type, {JOB_ID: job_id})
+        err = JobRequestException(JOBS_MISSING_ERR, [job_id])
+        check_id_error(job_comm, req_dict, err)
 
-        msg = self.jc._comm.last_message
-        assert msg == {
-            "msg_type": ERROR,
-            "content": {
-                "request": request,
-                "source": source,
-                **extra_params,
-                "name": type(err).__name__,
-                "message": str(err),
-            },
-        }
+    # run directly
+    with pytest.raises(type(err), match=re.escape(str(err))):
+        job_comm._msg_map[request_type](req)
 
-    def check_job_id_list__no_jobs(self, request_type):
-        job_id_list = [None, ""]
-        req_dict = make_comm_msg(request_type, job_id_list, False)
-        req = make_comm_msg(request_type, job_id_list, True)
-        err = JobRequestException(JOBS_MISSING_ERR, job_id_list)
 
-        # using handler
-        with pytest.raises(type(err), match=re.escape(str(err))):
-            self.jc._handle_comm_message(req_dict)
-        self.check_error_message(req_dict, err)
+def check_job_id__dne_test(job_comm: JobComm, request_type: str) -> None:
+    """Check that the correct error is raised with a job that does not exist."""
+    req_dict = make_comm_msg(request_type, {JOB_ID: JOB_NOT_FOUND})
+    err = JobRequestException(JOB_NOT_REG_ERR, JOB_NOT_FOUND)
+    check_id_error(job_comm, req_dict, err)
 
-        # run directly
-        with pytest.raises(type(err), match=re.escape(str(err))):
-            self.jc._msg_map[request_type](req)
 
-    def check_job_id_list__dne_jobs(self, request_type, response_type=None):
-        job_id_list = [None, "", JOB_NOT_FOUND, JOB_NOT_FOUND, BAD_JOB_ID]
-        expected_output = {
-            {"job_id": job_id, "error": generate_error(job_id, "not_found")}
-            for job_id in job_id_list
-            if job_id
-        }
-        req_dict = make_comm_msg(request_type, job_id_list, False)
-        self.jc._handle_comm_message(req_dict)
-        msg = self.jc._comm.last_message
-        assert msg == {
-            "msg_type": response_type if response_type else request_type,
-            "content": expected_output,
-        }
+def check_batch_id__no_job_test(job_comm: JobComm, request_type: str) -> None:
+    """Check that the correct error is raised with no batch job provided."""
+    for job_id in [None, ""]:
```

Comment:
ugly, but once the tests are moved out of the unittest class, this can be turned into test parameters